mteb/encoder_interface.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import Any, Protocol, runtime_checkable
+
+import numpy as np
+import torch
+
+
+@runtime_checkable
+class Encoder(Protocol):
+    """The interface for an encoder in MTEB."""
+
+    def encode(
+        self, sentences: list[str], prompt: str, **kwargs: Any
```

Comment:
Would prompts be `Optional[str]`, like in SentenceTransformer.encode?
```
def encode(
        self,
        sentences: Union[str, List[str]],
        prompt_name: Optional[str] = None,
        prompt: Optional[str] = None,
        ...
```

Same for `encode_queries` and `encode_corpus`.

Reply:
Ahh, this is great thanks! (especially prompt_name!)